### PR TITLE
AWS Kfdef changes using v3 stacks 

### PR DESCRIPTION
**Pull request by PatrickXYS**
**Which issue is resolved by this Pull Request:**
It helps AWS migrate to manifest V3.

**Description of your changes:**
1. Use V3 `kfctl_aws.yaml` and `kfctl_aws_congnito.yaml`.
2. Update some of aws components and create `stacks/aws/application` directory for a work-around.
3. Adopt AWS ECR image in `jupyter-web-app` component.
4. Adopt multi-user KFP.

**Checklist:**
- [x] Unit tests have been rebuilt: 
    1. `cd manifests/tests`
    2. `make generate-changed-only`
    3. `make test`

### CODE REVIEW

File: stacks/aws/kustomization.yaml [hunk 1]
```diff
@@ -31,7 +31,6 @@ resources:
   # AWS components
   - ../../aws/aws-alb-ingress-controller/base_v3
   - ../../aws/nvidia-device-plugin/overlays/application
-  - ../../aws/istio-ingress/base_v3
```
Comment (Jeffwan) [inline stacks/aws/kustomization.yaml:34]: So if we deploy this in istio namespace, there's no way to put it here, right?

File: kfdef/kfctl_aws.yaml [hunk 2]
```diff
@@ -7,417 +7,98 @@ spec:
...
+        path: stacks/aws/application/cluster-local-gateway-1-3-1
     name: cluster-local-gateway
   - kustomizeConfig:
-      parameters:
-      - name: clusterRbacConfig
-        value: 'ON'
       repoRef:
         name: manifests
-        path: istio/istio
+        path: istio/istio/base
```
Comment (Jeffwan) [inline kfdef/kfctl_aws.yaml:25]: em.. we probably want to create a stack for all istio manifests later

File: kfdef/kfctl_aws.yaml [hunk 3]
```diff
@@ -7,417 +7,98 @@ spec:
...
-    name: istio-crds
-  - kustomizeConfig:
-      parameters:
-      - name: namespace
-        value: istio-system
       repoRef:
         name: manifests
-        path: istio-1-3-1/istio-install-1-3-1
-    name: istio-install
+        path: stacks/aws/application/istio-1-3-1-stack
```
Comment (Jeffwan) [inline kfdef/kfctl_aws.yaml:15]: For long term, it's better for us to use v3 application in root folder and construct our sub-stack directly without extra application concepts. It looks fine now.

File: stacks/aws/kustomization.yaml (hunk 1 above)
Comment (PatrickXYS) [reply stacks/aws/kustomization.yaml:34]: Right, so this is kind-of work-around. I didn't come up with a better idea

File: kfdef/kfctl_aws.yaml (hunk 2 above)
Comment (PatrickXYS) [reply kfdef/kfctl_aws.yaml:25]: yeah makes sense, we can also do so for cert-manager later

File: kfdef/kfctl_aws.yaml (hunk 3 above)
Comment (PatrickXYS) [reply kfdef/kfctl_aws.yaml:15]: I think so, in the future, we need to get rid of these applications.

**Comment by Jeffwan** [comment]
@PatrickXYS  Make sure you follow this process and generate test case for new manifest change


- cd manifests/tests
- make generate-changed-only
- make test

File: stacks/aws/kustomization.yaml [hunk 4]
```diff
@@ -8,7 +8,7 @@ resources:
   - ../../admission-webhook/webhook/v3
   - ../../common/centraldashboard/overlays/stacks
   - ../../kubeflow-roles/base
-  - ../../jupyter/jupyter-web-app/base_v3
+  - ../../jupyter/jupyter-web-app/overlays/aws
```
Comment (Jeffwan) [inline stacks/aws/kustomization.yaml:11]: @PatrickXYS Can you have a separate application with aws overlay under /stack/aws without modifying the jupyter overlay?

File: pipeline/installs/multi-user/api-service/kustomization.yaml [hunk 5]
```diff
@@ -5,4 +5,5 @@ resources:
 - cluster-role.yaml
 configMapGenerator:
 - name: pipeline-api-server-config
-  env: params.env
+  envs:
+  - params.env
```
Comment (Jeffwan) [inline pipeline/installs/multi-user/api-service/kustomization.yaml:9]: Can we file a separate PR for this?

File: stacks/aws/kustomization.yaml (hunk 4 above)
Comment (PatrickXYS) [reply stacks/aws/kustomization.yaml:11]: That's the overlay we created before for ECR image. 

I'll make change in `/stack/aws`, in the future, we'll remove this overlay under `jupyter-web-app` component.

File: pipeline/installs/multi-user/api-service/kustomization.yaml (hunk 5 above)
Comment (PatrickXYS) [reply pipeline/installs/multi-user/api-service/kustomization.yaml:9]: Makes sense, will do

File: stacks/aws/kustomization.yaml (hunk 4 above)
Comment (PatrickXYS) [reply stacks/aws/kustomization.yaml]: Fixed.

File: pipeline/installs/multi-user/api-service/kustomization.yaml (hunk 5 above)
Comment (PatrickXYS) [reply pipeline/installs/multi-user/api-service/kustomization.yaml]: Fixed.

**Comment by Bobgy** [comment]
/lgtm
regarding KFP related references

**Comment by PatrickXYS** [comment]
Quick update: I tested against v3 Dex / Cognito / Multi-user KFP, all succeeded without any issue. 

Will leave this open until we make decision to merge. 

Related kfctl PR: https://github.com/kubeflow/kfctl/pull/380

**Comment by Jeffwan** [comment]
Excellent work! @PatrickXYS

**Pull request merged**